### PR TITLE
Add information about AutoSampler to the docs

### PR DESCRIPTION
**Pull request by y0z**
## Motivation & Description of the changes

Add information about [AutoSampler](https://hub.optuna.org/samplers/auto_sampler/) to the docs.

image

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@c-bata Could you review this PR?

File: docs/source/reference/samplers/index.rst [hunk 1]
```diff
@@ -11,6 +11,9 @@ The :mod:`~optuna.samplers` module defines a base class for parameter sampling a
 .. seealso::
     :ref:`user_defined_sampler` tutorial could be helpful if you want to implement your own sampler classes.
 
+.. seealso::
+    If you are unsure about which sampler to use, please consider using `AutoSampler <https://hub.optuna.org/samplers/auto_sampler/>`__, which automatically selects a promising sampler during optimization.
```
Comment (HideakiImamura) [inline docs/source/reference/samplers/index.rst:15]:
```suggestion
    If you are unsure about which sampler to use, please consider using `AutoSampler <https://hub.optuna.org/samplers/auto_sampler/>`__, which automatically selects a sampler during optimization.
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have a minor comment around wording.

**Review by c-bata — APPROVED** [review]
LGTM.

There are currently too many `seealso` sections on the page. This could be improved in a future pull request.

**Review by HideakiImamura — APPROVED** [review]
LGTM!

**Pull request merged**